R/mod_indicator.R

```diff
@@ -199,13 +202,63 @@ indicator_server <- function(id, registry_tracker, pool) {
       rv$ind_data$title <- input$ind_title
       rv$ind_data$short_description <- input$ind_short
       rv$ind_data$long_description <- input$ind_long
-      update_ind_text(pool, rv$ind_data)
-      rv$ind_data <- get_registry_ind(pool, input$indicator_registry) %>%
+      update_ind_text(pool_verify, rv$ind_data)
+      rv$ind_data <- get_registry_ind(pool_verify, input$indicator_registry) %>%
         dplyr::filter(.data$id == input$indicator)
     })
 
+    shiny::observeEvent(input$new_indicator, {
+      shinyalert::shinyalert(
+        title = "",
+        text = "Velg navn p\u00e5 ny indikator",
+        type = "input",
+        inputType = "text",
+        showCancelButton = TRUE,
+        callbackR = function(x) {
+
+          # Early guard (cancel button)
+          if (x == FALSE) {
+            return(NULL)
+          }
+
+          input_checks <- vector()
+
+          # Must be string
+          input_checks <- c(input_checks, is.character(x))
+          # Must have characters
+          input_checks <- c(input_checks, nchar(x) > 0)
+          # No special characters
+          input_checks <- c(input_checks, !grepl("[\u00e6\u00f8\u00e5]", x))
+          # No whitespace
+          input_checks <- c(input_checks, !grepl("[[:space:]]", x))
```

Comment:
Kanskje enklere med denne (sjekke at det kun er vanlige bokstaver, tall eller understrek)?
```
grepl("^[a-zA-Z0-9_]+$", string)
```

Reply:
Fikset. Vi bikker akkurat grensen for syklomatisk kompleksitet i lintingen på grunn av if-testen.